tests/components/linear_garage_door/test_light.py

```diff
@@ -0,0 +1,273 @@
+    """Test that data gets parsed and returned appropriately."""
+
+    await async_init_integration(hass)
+
+    assert hass.data[DOMAIN]
+    entries = hass.config_entries.async_entries(DOMAIN)
+    assert entries
+    assert len(entries) == 1
+    assert entries[0].state == ConfigEntryState.LOADED
+    assert hass.states.get("light.test_garage_1_light").state == STATE_ON
+    assert hass.states.get("light.test_garage_2_light").state == STATE_OFF
+
+
+async def test_turn_on(hass: HomeAssistant) -> None:
+    """Test that turning on the light works as intended."""
+
+    await async_init_integration(hass)
+
+    with (
+        patch(
+            "homeassistant.components.linear_garage_door.light.Linear.login",
+            return_value=True,
+        ),
+        patch(
+            "homeassistant.components.linear_garage_door.light.Linear.operate_device",
+            return_value=None,
+        ) as operate_device,
+        patch(
+            "homeassistant.components.linear_garage_door.light.Linear.close",
+            return_value=True,
+        ),
+    ):
+        await hass.services.async_call(
+            LIGHT_DOMAIN,
+            SERVICE_TURN_ON,
+            {ATTR_ENTITY_ID: "light.test_garage_2_light"},
+            blocking=True,
+        )
+
+    assert operate_device.call_count == 1
+    with (
+        patch(
+            "homeassistant.components.linear_garage_door.light.Linear.login",
+            return_value=True,
+        ),
+        patch(
+            "homeassistant.components.linear_garage_door.light.Linear.get_devices",
+            return_value=[
+                {
+                    "id": "test1",
+                    "name": "Test Garage 1",
+                    "subdevices": ["GDO", "Light"],
+                },
+                {
+                    "id": "test2",
+                    "name": "Test Garage 2",
+                    "subdevices": ["GDO", "Light"],
+                },
+            ],
+        ),
+        patch(
+            "homeassistant.components.linear_garage_door.light.Linear.get_device_state",
+            side_effect=lambda id: {
+                "test1": {
+                    "GDO": {"Open_B": "true", "Open_P": "100"},
+                    "Light": {"On_B": "true", "On_P": "100"},
+                },
+                "test2": {
+                    "GDO": {"Open_B": "false", "Open_P": "0"},
+                    "Light": {"On_B": "true", "On_P": "100"},
+                },
+                "test3": {
+                    "GDO": {"Open_B": "false", "Opening_P": "0"},
+                    "Light": {"On_B": "false", "On_P": "0"},
+                },
+                "test4": {
+                    "GDO": {"Open_B": "true", "Opening_P": "100"},
+                    "Light": {"On_B": "true", "On_P": "100"},
+                },
+            }[id],
+        ),
+        patch(
+            "homeassistant.components.linear_garage_door.light.Linear.close",
+            return_value=True,
+        ),
```

Comment:
Consider creating a broader patch, checkout analytics_insights or Epion

Reply:
Take a look at what I did with the patches, is that enough?